### PR TITLE
Fix aggressive font fallback on Java 11

### PR DESCRIPTION
**Pull request by miurahr**
- Fix [BUG#1051](https://sourceforge.net/p/omegat/bugs/1051/)
- Font#canDisplay, #canDisplayUpTo alternative methods
- Blacklist "Noto Color Emoji" from fallbacks

### CODE REVIEW

File: src/org/omegat/util/gui/FontFallbackListener.java [hunk 1]
```diff
@@ -64,10 +63,6 @@ public void insertUpdate(final DocumentEvent e) {
     }
 
     private void doStyling(Document document, final int offset, final int length) {
-        if (!Preferences.isPreference(Preferences.FONT_FALLBACK)) {
-            return;
-        }
-
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackListener.java:70]: Why was this removed?

File: src/org/omegat/util/gui/FontFallbackManager.java [hunk 2]
```diff
@@ -84,10 +191,12 @@ private static Font getCapableFontInternal(int cp) {
         for (int testIndex, i = 0; i < RECENT_FONTS.length; i++) {
             testIndex = (lastFontIndex - i + RECENT_FONTS.length) % RECENT_FONTS.length;
             Font font = RECENT_FONTS[testIndex];
-            if (font != null && font.canDisplay(cp)) {
-                lastFontIndex = testIndex;
-                CACHE.put(cp, font);
-                return font;
+            if (font != null) {
+                if (canDisplay(font, cp)) {
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackManager.java:195]: Why nest these conditions?

File: src/org/omegat/util/gui/FontFallbackManager.java [hunk 3]
```diff
@@ -52,15 +57,117 @@ private FontFallbackManager() {
...
+    private static final FontRenderContext DEFAULT_CONTEXT = new FontRenderContext(null, false, false);
+
+    static {
+        String[] fontFamilies = GraphicsEnvironment.getLocalGraphicsEnvironment().getAvailableFontFamilyNames();
+        for (final String fontName : fontFamilies) {
+            if (!fontName.endsWith(BOLD_SUFFIX) && !fontName.endsWith(ITALIC_SUFFIX)) {
+                myFontFamilyNames.add(fontName);
+            }
+        }
+    }
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackManager.java:84]: A couple things:

1. I don't understand why this moved to `FontFallbackManager`. Callers of `StaticUtils#getFontNames` are not interested in font fallback, and neither `FontFallbackManager#getFontNames` nor `myFontFamilyNames` are used within `FontFallbackManager` at all. I think it should be moved back to `StaticUtils`.
2. None of the callers of `getFontNames` are in a hot loop, so I am skeptical that it is meaningful performance-wise to cache the family names. In fact by caching the names but never invalidating the cache, you prevent new fonts from being picked up (I don't actually know if this API actually can pick up newly installed fonts, but if it can then your caching scheme definitely breaks that).
3. What is the `BOLD_SUFFIX` and `ITALIC_SUFFIX`? For reference on my machine (macOS 11.4) none of my fonts have such suffixes:

    <details><summary>Aaron's fonts</summary>
    [.AppleSystemUIFont, Academy Engraved LET, Agave, Al Bayan, Al Nile, Al Tarikh, Alegreya Sans, Alegreya Sans SC, all-the-icons, American Typewriter, Andale Mono, Anonymous Pro, Apple Braille, Apple Chancery, Apple Color Emoji, Apple LiGothic, Apple LiSung, Apple SD Gothic Neo, Apple Symbols, AppleGothic, AppleMyungjo, Arial, Arial Black, Arial Hebrew, Arial Hebrew Scholar, Arial Narrow, Arial Rounded MT Bold, Arial Unicode MS, Avenir, Avenir Next, Avenir Next Condensed, Ayuthaya, BabelStone Han, BabelStone Han PUA, Baghdad, Baloo, Baloo Bhai, Baloo Bhaijaan, Baloo Bhaina, Baloo Chettan, Baloo Da, Baloo Paaji, Baloo Tamma, Baloo Tammudu, Baloo Thambi, Bangla MN, Bangla Sangam MN, Baoli SC, Baoli TC, Baskerville, Beirut, BiauKai, Big Caslon, Bodoni 72, Bodoni 72 Oldstyle, Bodoni 72 Smallcaps, Bodoni Ornaments, Bradley Hand, Brush Script MT, Cambay Devanagari, Cascadia Code, Cascadia Code Italic, Cascadia Code PL, Cascadia Code PL Italic, Cascadia Mono, Cascadia Mono Italic, Cascadia Mono PL, Cascadia Mono PL Italic, Chalkboard, Chalkboard SE, Chalkduster, Charter, Cochin, Comic Sans MS, Copperplate, Corsiva Hebrew, Courier, Courier New, Damascus, DecoType Naskh, Devanagari MT, Devanagari Sangam MN, Dialog, DialogInput, Didot, DIN Alternate, DIN Condensed, Diwan Kufi, Diwan Thuluth, EB Garamond, EB Garamond 12 All SC, EB Garamond Initials, EB Garamond Initials Fill1, EB Garamond Initials Fill2, EB Garamond SC, Envy Code R, ETBembo, ETBookOT, Euphemia UCAS, Farah, Farisi, file-icons, Fira Code, FiraGO, FontAwesome, Futura, Galvji, GB18030 Bitmap, Geeza Pro, Geneva, Georgia, Gill Sans, github-octicons, Go, Go Medium, Go Mono, Go Smallcaps, Gotu, Grantha Sangam MN, Gujarati MT, Gujarati Sangam MN, GungSeo, Gurmukhi MN, Gurmukhi MT, Gurmukhi Sangam MN, Hack, HanaMinA, HanaMinB, Hangulatin EN, Hannotate SC, Hannotate TC, HanziPen SC, HanziPen TC, HeadLineA, Hei, Heiti SC, Heiti TC, Helvetica, Helvetica Neue, Herculanum, Hiragino Maru Gothic ProN, Hiragino Mincho ProN, Hiragino Sans, Hiragino Sans CNS, Hiragino Sans GB, Hoefler Text, IBM 3270, IBM 3270 Condensed, IBM 3270 Semi-Condensed, IBM Plex Mono, IBM Plex Sans, IBM Plex Sans Arabic, IBM Plex Sans Condensed, IBM Plex Sans Devanagari, IBM Plex Sans Hebrew, IBM Plex Sans KR, IBM Plex Sans Thai, IBM Plex Sans Thai Looped, IBM Plex Serif, Impact, InaiMathi, Inconsolata, Input Mono, Input Mono Compressed, Input Mono Condensed, Input Mono Narrow, Input Sans, Input Sans Compressed, Input Sans Condensed, Input Sans Narrow, Input Serif, Input Serif Compressed, Input Serif Condensed, Input Serif Narrow, Inter, Iosevka, ITF Devanagari, ITF Devanagari Marathi, Jaini, Jaini Purva, JetBrains Mono, JetBrains Mono NL, JuliaMono, Kai, Kailasa, Kaiti SC, Kaiti TC, Kannada MN, Kannada Sangam MN, Kefa, Khmer MN, Khmer Sangam MN, Klee, Kohinoor Bangla, Kohinoor Devanagari, Kohinoor Gujarati, Kohinoor Telugu, Kokonor, Krungthep, KufiStandardGK, Lahore Gurmukhi, Lantinghei SC, Lantinghei TC, Lao MN, Lao Sangam MN, Libian SC, Libian TC, Ligconsolata, LiHei Pro, LingWai SC, LingWai TC, LiSong Pro, Lucida Grande, Luminari, M+ 1c, M+ 1m, M+ 1mn, M+ 1p, M+ 2c, M+ 2m, M+ 2p, Maku, Malayalam MN, Malayalam Sangam MN, Marker Felt, Material Icons, MD IO 0.2, Menlo, Meta, Microsoft Sans Serif, MigMix 1M, MigMix 1P, MigMix 2M, MigMix 2P, Migu 1C, Migu 1M, Migu 1P, Migu 2M, Mishafi, Mishafi Gold, Modak, Monaco, mononoki, Monospaced, Mshtakan, Mukta, Mukta Mahee, Mukta Malar, Mukta Vaani, Muna, Myanmar MN, Myanmar Sangam MN, Nadeem, Nanum Brush Script, Nanum Gothic, Nanum Myeongjo, Nanum Pen Script, Neue Haas Grotesk Text Pro, New Peninim MT, Noteworthy, Noto Nastaliq Urdu, Noto Sans Kannada, Noto Sans Myanmar, Noto Sans Oriya, Noto Serif Kannada, Noto Serif Myanmar, Optima, Oriya MN, Oriya Sangam MN, Osaka, Palatino, Papyrus, Party LET, PCMyungjo, Phosphate, Piazzolla, Piazzolla SC, PilGi, PingFang HK, PingFang SC, PingFang TC, Plantagenet Cherokee, PragmataPro, PragmataPro Liga, PragmataPro Mono, PragmataPro Mono Liga, PSL Ornanong Pro, PT Mono, PT Sans, PT Sans Caption, PT Sans Narrow, PT Serif, PT Serif Caption, Raanana, Rockwell, Sana, SansSerif, Sarasa Fixed CL, Sarasa Fixed HC, Sarasa Fixed J, Sarasa Fixed K, Sarasa Fixed SC, Sarasa Fixed Slab CL, Sarasa Fixed Slab HC, Sarasa Fixed Slab J, Sarasa Fixed Slab K, Sarasa Fixed Slab SC, Sarasa Fixed Slab TC, Sarasa Fixed TC, Sarasa Gothic CL, Sarasa Gothic HC, Sarasa Gothic J, Sarasa Gothic K, Sarasa Gothic SC, Sarasa Gothic TC, Sarasa Mono CL, Sarasa Mono HC, Sarasa Mono J, Sarasa Mono K, Sarasa Mono SC, Sarasa Mono Slab CL, Sarasa Mono Slab HC, Sarasa Mono Slab J, Sarasa Mono Slab K, Sarasa Mono Slab SC, Sarasa Mono Slab TC, Sarasa Mono TC, Sarasa Term CL, Sarasa Term HC, Sarasa Term J, Sarasa Term K, Sarasa Term SC, Sarasa Term Slab CL, Sarasa Term Slab HC, Sarasa Term Slab J, Sarasa Term Slab K, Sarasa Term Slab SC, Sarasa Term Slab TC, Sarasa Term TC, Sarasa UI CL, Sarasa UI HC, Sarasa UI J, Sarasa UI K, Sarasa UI SC, Sarasa UI TC, Sathu, Savoye LET, Sawarabi Mincho, Serif, SF Compact, SF Compact Display, SF Compact Rounded, SF Compact Text, SF Pro, SF Pro Display, SF Pro Rounded, SF Pro Text, Shobhika, Shree Devanagari 714, Signifier Test, SignPainter, Silom, SimSong, Sinhala MN, Sinhala Sangam MN, Skia, Snell Roundhand, Songti SC, Songti TC, Source Sans 3, STFangsong, STHeiti, STIXGeneral, STIXIntegralsD, STIXIntegralsSm, STIXIntegralsUp, STIXIntegralsUpD, STIXIntegralsUpSm, STIXNonUnicode, STIXSizeFiveSym, STIXSizeFourSym, STIXSizeOneSym, STIXSizeThreeSym, STIXSizeTwoSym, STIXVariants, STKaiti, STSong, Sudo, Sudo UI, Sudo Var, Sukhumvit Set, Symbol, Tahoma, Tamil MN, Tamil Sangam MN, Telugu MN, Telugu Sangam MN, Thonburi, Times, Times New Roman, Tiro Bangla, Tiro Gurmukhi, Tiro Hindi, Tiro Kannada, Tiro Marathi, Tiro Sanskrit, Tiro Tamil, Tiro Telugu, Toppan Bunkyu Gothic, Toppan Bunkyu Midashi Gothic, Toppan Bunkyu Midashi Mincho, Toppan Bunkyu Mincho, Trattatello, Trebuchet MS, Tsukushi A Round Gothic, Tsukushi B Round Gothic, Unifont, Unifont-JP, Valkyrie C3, Valkyrie C3 Tab, Valkyrie C4, Valkyrie C4 Tab, Valkyrie T3, Valkyrie T3 Tab, Valkyrie T4, Valkyrie T4 Tab, Verdana, Victor Mono, Waseem, Wawati SC, Wawati TC, Weather Icons, Webdings, Weibei SC, Weibei TC, Wingdings, Wingdings 2, Wingdings 3, Xingkai SC, Xingkai TC, Yuanti SC, Yuanti TC, YuGothic, YuKyokasho, YuKyokasho Yoko, YuMincho, YuMincho +36p Kana, Yuppy SC, Yuppy TC, Zapf Dingbats, Zapfino]
    </details>

**Review by amake — COMMENTED** [review]
Thanks, this does make fallback work for me on Java 11, but it seems like it's quite slow. I'd like to do some benchmarking, especially since this changes the codepath for macOS on Java 8 as well (which wasn't broken, so it's a regression for it to get slower).

Source file

```
emoji-1.0:  😀😃😇
Unicode 6.1：😀😗😙😑😮😯😴😛😕😟
Unicode 7.0：🙂🙁🕵🗣🕴🖕🖖🖐
Unicode 8.0：🤗🤓🤔🙄🤐🙃🤑🤒🤕🤖
Unicode 9.0：🤣🤠🤡🤥🤤🤢🤧🤴🤶🤵🤷
Unicode 10.0：🤩🤨🤯🤪🤬🤮🤫🤭🧐🧒
Others: 𓁲𝒢
```

Log (Java 11, this PR: 67917b)

```
21445: FINE: Searching 2167 fonts for one supporting U+1f600 😀 
21445: FINE: Search found BabelStone Han in 12169 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f575 🕵 
21445: FINE: Search failed to find a font; time: 16685 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f5e3 🗣 
21445: FINE: Search failed to find a font; time: 9 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f574 🕴 
21445: FINE: Search failed to find a font; time: 9 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f595 🖕 
21445: FINE: Search found HanaMinA in 4 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f596 🖖 
21445: FINE: Search failed to find a font; time: 7 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f936 🤶 
21445: FINE: Search failed to find a font; time: 10 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f935 🤵 
21445: FINE: Search failed to find a font; time: 9 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f937 🤷 
21445: FINE: Search failed to find a font; time: 9 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1f9d2 🧒 
21445: FINE: Search failed to find a font; time: 10 ms 
21445: FINE: Searching 2167 fonts for one supporting U+13072 𓁲 
21445: FINE: Search failed to find a font; time: 9 ms 
21445: FINE: Searching 2167 fonts for one supporting U+1d4a2 𝒢 
21445: FINE: Search found JuliaMono in 5 ms 
```

Log (Java 8, this PR: 67917b)

```
13952: FINE: Searching 2167 fonts for one supporting U+1f600 😀 
13952: FINE: Search found BabelStone Han in 8616 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f575 🕵 
13952: FINE: Search failed to find a font; time: 19549 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f5e3 🗣 
13952: FINE: Search failed to find a font; time: 80 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f574 🕴 
13952: FINE: Search failed to find a font; time: 75 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f595 🖕 
13952: FINE: Search found HanaMinA in 29 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f596 🖖 
13952: FINE: Search failed to find a font; time: 69 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f936 🤶 
13952: FINE: Search failed to find a font; time: 122 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f935 🤵 
13952: FINE: Search failed to find a font; time: 84 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f937 🤷 
13952: FINE: Search failed to find a font; time: 74 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1f9d2 🧒 
13952: FINE: Search failed to find a font; time: 85 ms 
13952: FINE: Searching 2167 fonts for one supporting U+13072 𓁲 
13952: FINE: Search failed to find a font; time: 105 ms 
13952: FINE: Searching 2167 fonts for one supporting U+1d4a2 𝒢 
13952: FINE: Search found JuliaMono in 51 ms 
```

Log (Java 8, master: 7834b1)
```
69145: FINE: Searching 2167 fonts for one supporting U+1f600 😀 
69145: FINE: Fetching latest version info 
69145: FINE: Fetched latest version info: {version=5.5.0, revision=dev, beta=_Beta, update=0} 
69145: FINE: OmegaT is up to date 
69145: FINE: Search found BabelStone Han in 6733 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f575 🕵 
69145: Info: Event: new entry activated (LOG_INFO_EVENT_ENTRY_ACTIVATED)
69145: FINER: Time for find matches: 24 
69145: FINE: Search failed to find a font; time: 17348 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f5e3 🗣 
69145: FINE: Search failed to find a font; time: 5 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f574 🕴 
69145: FINE: Search failed to find a font; time: 6 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f595 🖕 
69145: FINE: Search found HanaMinA in 4 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f596 🖖 
69145: FINE: Search failed to find a font; time: 4 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f936 🤶 
69145: FINE: Search failed to find a font; time: 5 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f935 🤵 
69145: FINE: Search failed to find a font; time: 3 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f937 🤷 
69145: FINE: Search failed to find a font; time: 4 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1f9d2 🧒 
69145: FINE: Search failed to find a font; time: 3 ms 
69145: FINE: Searching 2167 fonts for one supporting U+13072 𓁲 
69145: FINE: Search failed to find a font; time: 4 ms 
69145: FINE: Searching 2167 fonts for one supporting U+1d4a2 𝒢 
69145: FINE: Search found JuliaMono in 3 ms 
```

File: src/org/omegat/util/gui/FontFallbackManager.java [hunk 4]
```diff
@@ -52,15 +57,117 @@ private FontFallbackManager() {
      * <li><a href="https://github.com/googlei18n/noto-emoji">Noto Emoji</a>
      * </ul>
      */
-    private static final Set<String> FONT_BLACKLIST = Collections.singleton("Apple Color Emoji");
-    private static final Font FONT_UNAVAILABLE = new Font("", 0, 0);
+    private static final Set<String> FONT_BLACKLIST = new HashSet<>(
+            Arrays.asList("Apple Color Emoji", "Noto Color Emoji")
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackManager.java:61]: Did you actually test Noto Color Emoji, or are you basing this on my comment on SourceForge?

Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackManager.java:59]: I tested on Linux.

File: src/org/omegat/util/gui/FontFallbackManager.java [hunk 5]
```diff
@@ -52,15 +55,97 @@ private FontFallbackManager() {
...
 
+    private static final FontRenderContext DEFAULT_CONTEXT = new FontRenderContext(null, false, false);
+
+    private FontFallbackManager() {
+    }
+
+    /**
+     * Detect specified codePoints can display with specified font.
+     */
+    public static int canDisplayUpTo(Font font, String str) {
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackManager.java:77]: It looks like you copied these implementations from the `Font` class. That's fine if that's really the only way to do it, but it would be good to mention that fact in the comments.

Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackManager.java:77]: Could you propose another way to improve here? If FontFallbackListener don't use canDisplayUpTo method, it can be simple. I don't have idea around here just now.

File: src/org/omegat/util/gui/FontFallbackManager.java (hunk 3 above)
Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackManager.java]: > None of the callers of getFontNames are in a hot loop,
This is regression when porting from my local dev branch to here.

> What is the BOLD_SUFFIX and ITALIC_SUFFIX?

These idea come from intellij-community source. I also don't know in my platform. So i'd like to remove it.

File: src/org/omegat/util/gui/FontFallbackManager.java [hunk 6]
```diff
@@ -52,15 +55,97 @@ private FontFallbackManager() {
...
+     * Detect specified codePoint can display with specified font.
+     * @param font check against.
+     * @param codePoint character to display.
+     * @return true when specifed character can display on font, otherwise false.
+     */
+    public static boolean canDisplay(Font font, final int codePoint) {
+        if (!Character.isValidCodePoint(codePoint)) return false;
+        if (Platform.isMacOSX()) {
+            int glyphCode = font.createGlyphVector(DEFAULT_CONTEXT, new String(new int[]{codePoint}, 0, 1)).getGlyphCode(0);
+            return (0 < glyphCode && glyphCode <= 0x00ffffff);
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackManager.java:143]: Instead of instantiating a string, you can use the method that takes an `int[]`.

Minor nitpicks:
- It's OmegaT style to always use brackets even for single-line bodies of `if` statements
- Unnecessary parens
```suggestion
        if (!Character.isValidCodePoint(codePoint)){
            return false;
        }
        if (Platform.isMacOSX()) {
            int glyphCode = font.createGlyphVector(DEFAULT_CONTEXT, Character.toChars(codePoint), 0, 1)).getGlyphCode(0);
            return 0 < glyphCode && glyphCode <= 0x00ffffff;
```

File: src/org/omegat/util/gui/FontFallbackManager.java (hunk 5 above)
Comment (amake) [reply src/org/omegat/util/gui/FontFallbackManager.java:77]: Yeah I don't have a good idea to improve it.

My concern with duplicating standard library code is that we could miss improvements or changes across Java versions.

Maybe it would be good to only use our version when the OS is Mac:

```java
if (!Platform.isMacOSX()) {
    return font.canDisplayUpTo(...);
} else {
    // This implementation is copied from java.awt.Font; we use it to work
    // around a bug on macOS whereby all fonts always claim to support all
    // codepoints; see bug#1051 (https://sourceforge.net/p/omegat/bugs/1051/)
    ...
}

```

File: src/org/omegat/util/gui/FontFallbackManager.java (hunk 6 above)
Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackManager.java]: merge the suggested.

File: src/org/omegat/util/gui/FontFallbackManager.java (hunk 2 above)
Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackManager.java]: fixed.

File: src/org/omegat/util/gui/FontFallbackListener.java (hunk 1 above)
Comment (miurahr) [reply src/org/omegat/util/gui/FontFallbackListener.java]: reverted.

File: src/org/omegat/util/gui/FontFallbackListener.java [hunk 7]
```diff
@@ -43,6 +44,7 @@
 
 import org.omegat.util.Preferences;
 
+
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackListener.java:47]:
```suggestion
```

File: src/org/omegat/util/gui/FontFallbackListener.java [hunk 8]
```diff
@@ -67,7 +69,6 @@ private void doStyling(Document document, final int offset, final int length) {
         if (!Preferences.isPreference(Preferences.FONT_FALLBACK)) {
             return;
         }
```
Comment (amake) [inline src/org/omegat/util/gui/FontFallbackListener.java:71]:
```suggestion
        }

```

File: src/org/omegat/util/gui/FontFallbackManager.java (hunk 5 above)
Comment (amake) [reply src/org/omegat/util/gui/FontFallbackManager.java:96]: I realized that we can be smarter about when we use the workaround code; see https://github.com/omegat-org/omegat/pull/118/commits/c7e3b3bdbd9c92b4cfd8d490cfc38d96522c25c4

**Comment by amake** [comment]
I made some changes; if everything looks OK to you then I will squash to `master`.

**Comment by miurahr** [comment]
I think it is ready for merge.

**Comment by amake** [comment]
Thanks for all your work on this. I've squashed to `master`.
